core/API/Request.php

```diff
@@ -269,7 +271,9 @@ public function process()
                 return $response->getResponse($returnedValue, $module, $method);
             });
         } catch (Exception $e) {
-            Log::debug($e);
+            StaticContainer::get(LoggerInterface::class)->error('Uncaught exception in API: {exception}', [
```

Comment:
I wonder if we should keep `debug` level when it is fetched through UI to not expose the specific error here? just not sure if there could be any problem such as sensitive information or something. Likely not a problem though.

Reply:
If it's not debug, it won't trigger the fingers crossed handler, BUT I made sure the `screen` handler doesn't get wrapped in the fingers crossed handler. So I think this will result in an exception displaying on the screen for UI requests, but it should just print the exception message, which is the current behavior right? I'll make sure to see exactly what happens.

Reply:
Was just not sure whether it could have any side effects re the exception message re security or whether it could be any annoying or cause any other problems. Wonder if it would even show during an API request etc or what would happen then? I wonder if we need to add tests for all those cases where this error is triggered.

Reply:
I meant mainly to make sure that the API output is still correct when an error occurs during an API request. Might be even good for one UI test to see what it shows there. Wouldn't need to check re what is logged to the file.

Reply:
> I meant mainly to make sure that the API output is still correct when an error occurs during an API request.

There are currently plenty of API tests that trigger errors, so think this one is already good.

> Might be even good for one UI test to see what it shows there. 

Will try to include a UI test. I guess the existing safemode one is good for the fatal error check.

Reply:
Created a UI test and noticed that the error will also appear as a notification. Not sure what the best way to handle this is, since w/o the log statement, there ends up being no record of the error in the file.

My personal opinion is that there really shouldn't be a screen logger, and everything should go into a log that only admins/devs can see. But I'm not sure if that change can be made.

Reply:
That's what I meant with changing the log level from debug to error. The screen logger is definitely useful in that it helps us finding bugs/issues etc. where otherwise 98% of users would never look in the logs and not even know they exist etc. 

The only thing I can think of be to change the log level depending on whether we are CLI or not. Or also enable the fingercrossedhandler on cli based on some other flags maybe. Can we maybe log it as `debug` and somehow pass some option to also attach all previous logs when this happens? Defeats the idea of the fingercrossedhandler maybe a little but could work maybe?

Reply:
Sure it should be fine this way as well 👍

Reply:
Made the change + added some more logs. Also modified ExceptionToTextProcessor to replace `{exception}` in the message instead of the whole message.